lib/facter/ethtool.rb

```diff
@@ -0,0 +1,75 @@
+module Ethtool
+  module Facts
+
+    # Check whether ethtool exists
+    def self.exists?
+      File.exists?('/sbin/ethtool')
+    end
+
+    # Run ethtool on an interface
+    def self.ethtool interface
+      %x{/sbin/ethtool #{interface} 2>/dev/null}
+    end
+
+    # Get all interfaces on the system
+    def self.interfaces
+      Dir.foreach('/sys/class/net').reject{|x| x.start_with?('.', 'veth')}
```

Comment:
this could be cached in instance var to avoid hitting file system multiple times. just throw in `@interfaces ||= ` in there.